dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/bundle/DefaultTrackerBundleService.java

```diff
@@ -152,6 +158,26 @@ public TrackerBundleReport commit( TrackerBundle bundle )
         return bundleReport;
     }
 
+    @Override
+    public void postCommit( TrackerBundle bundle )
+    {
+        updateTeisLastUpdated( bundle );
+    }
+
+    private void updateTeisLastUpdated( TrackerBundle bundle )
+    {
+        if ( Optional.ofNullable( bundle.getUpdatedTeis() ).filter( ut -> !ut.isEmpty() ).isPresent() )
+        {
+            List<List<String>> uidsPartitions = Lists.partition( Lists.newArrayList( bundle.getUpdatedTeis() ), 20000 );
```

Comment:
I was wondering if this partitioning logic can be moved to inside the store method. That way, whoever uses the store method need not worry about partitioning. 
In this way, the partitioning responsibility is left to the caller and higher chances of it being missed.

Reply:
yes good point thanks